Log images: wrong orientation and missing coordinate

Reported by a fellow geocacher, reproducible for me.

Cache in question: https://coord.info/GC3F6BT

When I load this cache in current release/master branch version (2025.05.26-aa748c6 developer build, 2025.05.26-d88eb6b developer build) or version 2025.05.14, I get:
* Only one listing image -> NOT OK? Images of cache gallery are missing
* One image of logs -> ok, correct orientation and lat/lon-coordinate in image metadata is shown

![Image](https://github.com/user-attachments/assets/06e28051-7d78-462e-89a9-631d70e1d1a8)

When I load the same cache in current beta version 2025.05.16-RC, I get:
* 5 images in listing -> O-ish; here loading of gallery is included but orientations are wrong
* One image of logs -> NOT OK: image is of wrong orientation and lat/lon-coordinate from metadata is gone

![Image](https://github.com/user-attachments/assets/dfa7f2d8-5426-445a-acec-a4d83a3748e5)

Gallery on web page looks fine:

![Image](https://github.com/user-attachments/assets/6ff17aec-1519-4447-84e2-5bce28d00cf6)

Something strange is happening here. Any idea?

Issue of not loaded gallery is caused by the fix of #16778, I reopened this issue.

Displaying of images in wrong orientation and with missing embedded coordinates has the same cause: in some cases (for example on my own mobile phone) image metadata can't be loaded. (Orientation is also part of this metadata). 

Here is a typical example exceptions (I have many of those in my logs):

```
06-22 19:15:28.951 15935 15962 W cgeo    : java.lang.RuntimeException: java.lang.InstantiationException: java.lang.Class<com.drew.metadata.exif.ExifIFD0Directory> cannot be instantiated
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.metadata.tiff.DirectoryTiffHandler.pushDirectory(SourceFile:65)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.metadata.exif.ExifTiffHandler.setTiffMarker(SourceFile:79)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.imaging.tiff.TiffReader.processTiff(SourceFile:65)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.metadata.exif.ExifReader.extract(SourceFile:94)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.metadata.exif.ExifReader.extract(SourceFile:84)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.metadata.exif.ExifReader.readJpegSegments(SourceFile:63)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.imaging.jpeg.JpegMetadataReader.processJpegSegmentData(SourceFile:134)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.imaging.jpeg.JpegMetadataReader.process(SourceFile:126)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.imaging.jpeg.JpegMetadataReader.readMetadata(SourceFile:77)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.imaging.jpeg.JpegMetadataReader.readMetadata(SourceFile:84)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.imaging.ImageMetadataReader.readMetadata(SourceFile:146)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.imaging.ImageMetadataReader.readMetadata(SourceFile:124)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.imaging.ImageMetadataReader.readMetadata(SourceFile:103)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at cgeo.geocaching.utils.MetadataUtils.readImageMetadata(SourceFile:57)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at cgeo.geocaching.network.HtmlImage.loadCachedImage(SourceFile:509)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at cgeo.geocaching.network.HtmlImage.loadCachedImage(SourceFile:475)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at cgeo.geocaching.network.HtmlImage.loadImageFromStorage(SourceFile:424)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at cgeo.geocaching.network.HtmlImage.access$200(SourceFile:57)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at cgeo.geocaching.network.HtmlImage$1.loadFromDisk(SourceFile:300)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at cgeo.geocaching.network.HtmlImage$1.$r8$lambda$8RnNeTZNb5HMlbBRwbRiIQUh-t4(SourceFile:283)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at cgeo.geocaching.network.HtmlImage$1$$ExternalSyntheticLambda3.run(SourceFile:0)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at io.reactivex.rxjava3.internal.schedulers.ScheduledDirectTask.call(SourceFile:38)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at io.reactivex.rxjava3.internal.schedulers.ScheduledDirectTask.call(SourceFile:25)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:307)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at java.lang.Thread.run(Thread.java:1012)
06-22 19:15:28.951 15935 15962 W cgeo    : Caused by: java.lang.InstantiationException: java.lang.Class<com.drew.metadata.exif.ExifIFD0Directory> cannot be instantiated
06-22 19:15:28.951 15935 15962 W cgeo    : 	at java.lang.Class.newInstance(Native Method)
06-22 19:15:28.951 15935 15962 W cgeo    : 	at com.drew.metadata.tiff.DirectoryTiffHandler.pushDirectory(SourceFile:63)
06-22 19:15:28.951 15935 15962 W cgeo    : 	... 27 more
```

I have no explanation yet why instantiation of class `com.drew.metadata.exif.ExifIFD0Directory` fails in this case.

--- a/main/proguard-project.txt
+++ b/main/proguard-project.txt
@@ -146,3 +146,6 @@
 
 # transitive needed using R8 (minifyEnabled)
 -dontwarn org.slf4j.impl.StaticLoggerBinder
+
+# keep classes of metadata libaray (this lib heavily uses reflection)
+-keep class com.drew.** { *; }

--- a/main/src/main/java/cgeo/geocaching/network/HtmlImage.java
+++ b/main/src/main/java/cgeo/geocaching/network/HtmlImage.java
@@ -507,6 +507,7 @@ public class HtmlImage implements Html.ImageGetter {
                 return ImmutableTriple.of(image, null, true);
             }
             metadata = MetadataUtils.readImageMetadata("[HtmlImage]" + uri, imageStream, true);
+            Log.iForce("[HtmlImage] Orientation of " + uri + ": " + MetadataUtils.getOrientation(metadata));
         }
         return ImmutableTriple.of(image, metadata, freshEnough);
     }

--- a/main/src/main/java/cgeo/geocaching/utils/MetadataUtils.java
+++ b/main/src/main/java/cgeo/geocaching/utils/MetadataUtils.java
@@ -54,7 +54,9 @@ public final class MetadataUtils {
             return null;
         }
         try {
-            return ImageMetadataReader.readMetadata(imageStream);
+            final Metadata data = ImageMetadataReader.readMetadata(imageStream);
+            //throw new RuntimeException("test");
+            return data;
         } catch (IOException | ImageProcessingException | RuntimeException ie) {
             Log.w("[MetadataUtils] Problem reading metadata from " + description, ie);
         } finally {
@@ -84,7 +86,7 @@ public final class MetadataUtils {
     }
 
     public static int getOrientation(final Metadata metadata) {
-        return safeProcess("orientation", metadata, ExifInterface.ORIENTATION_NORMAL, () -> {
+        return safeProcess("orientation", metadata, ExifInterface.ORIENTATION_UNDEFINED, () -> {
 
             final Collection<ExifDirectoryBase> exifDirs = metadata.getDirectoriesOfType(ExifDirectoryBase.class);
             for (ExifDirectoryBase exifDir : exifDirs) {

